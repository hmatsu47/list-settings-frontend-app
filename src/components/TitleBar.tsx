import AppBar from "@suid/material/AppBar";
import Box from "@suid/material/Box";
import IconButton from "@suid/material/IconButton";
import Stack from "@suid/material/Stack";
import Typography from "@suid/material/Typography";
import ReplayIcon from "@suid/icons-material/Replay";
import {
  fetchLocal,
  fetchRemote,
  fetchUriSettings,
} from "../api/fetchUriSettings";
import { fetchTagSettings } from "../api/fetchTagSettings";

const reload = () => {
  fetchUriSettings(fetchLocal);
  fetchUriSettings(fetchRemote);
  fetchTagSettings();
};

export const TitleBar = () => {
  return (
    <AppBar position="static">
      <Stack direction="row">
        <Typography
          variant="h6"
          component="div"
          sx={{
            flexGrow: 1,
            padding: "0 0 0 10px",
          }}
        >
          コンテナリリースイメージ一覧
        </Typography>
        <Box title="再読み込み">
          <IconButton
            size="small"
            edge="start"
            color="inherit"
            aria-label="reload"
            sx={{ mr: 1 }}
            onClick={() => reload()}
          >
            <ReplayIcon sx={{ fontSize: "large", padding: "2px" }} />
            Reload
          </IconButton>
        </Box>
      </Stack>
    </AppBar>
  );
};
