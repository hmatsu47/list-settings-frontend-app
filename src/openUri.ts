export const openUriInNewTab = (uri: string) => {
  window.open(uri, "_blank", "noopener,noreferrer");
};