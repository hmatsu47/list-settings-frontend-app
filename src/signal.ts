import { createSignal } from "solid-js";
import { AlertColor } from "@suid/material/Alert";

import { TagSetting, UriSetting } from "./type";

export const [services, setServices] =
  createSignal<string[] | null | undefined>();
export const [remoteServices, setRemoteServices] =
  createSignal<string[] | null | undefined>();
export const [tagSettings, setTagSettings] =
  createSignal<TagSetting[] | null | undefined>();
export const [uriSettings, setUriSettings] =
  createSignal<UriSetting[] | null | undefined>();
export const [uriRemoteSettings, setUriRemoteSettings] =
  createSignal<UriSetting[] | null | undefined>();
export const [message, setMessage] = createSignal<string | undefined>();
export const [messageSeverity, setMessageSeverity] =
  createSignal<AlertColor | undefined>();
