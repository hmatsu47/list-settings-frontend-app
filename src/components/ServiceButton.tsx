import Button from "@suid/material/Button";
import { service, setService } from "../signal";

type Props = {
  serviceName: string;
};

export const ServiceButton = (props: Props) => {
  return (
    <Button
      variant={service() === props.serviceName ? "contained" : "outlined"}
      size="small"
      color={service() === props.serviceName ? "primary" : "inherit"}
      onClick={(e) => {
        setService(props.serviceName);
        // 選択肢をローカルストレージに記録しておく
        localStorage.setItem("selectedService", props.serviceName);
      }}
      sx={{ textTransform: "none" }}
      title={`${props.serviceName} を選択`}
    >
      {props.serviceName}
    </Button>
  );
};
