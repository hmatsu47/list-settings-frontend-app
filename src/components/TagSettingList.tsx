import { For, Show } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Paper from "@suid/material/Paper";
import Stack from "@suid/material/Stack";
import Table from "@suid/material/Table";
import TableBody from "@suid/material/TableBody";
import TableCell from "@suid/material/TableCell";
import TableContainer from "@suid/material/TableContainer";
import TableHead from "@suid/material/TableHead";
import TableRow from "@suid/material/TableRow";
import Typography from "@suid/material/Typography";
import { openUriInNewTab } from "../openUri";
import { tagSettings } from "../signal";

const convertEnvironmentName = (environmentName: string) => {
  switch (environmentName) {
    case "dev":
      return "開発";
    case "prod":
      return "本番";
  }
  return environmentName;
};

const buttonColor = (environmentName: string) => {
  const colors = localStorage.getItem("tagButtonColor");
  if (!colors) {
    return "#616161";
  }
  const map = new Map<string, string>(Object.entries(JSON.parse(colors)));
  return map.get(environmentName) ?? "#616161";
};

export const TagSettingList = () => {
  const headerTitle = localStorage.getItem("tagSettingsHeaderTitle");
  const uriPrefix = localStorage.getItem("tagSettingUriPrefix");
  const uriSuffix = localStorage.getItem("tagSettingUriSuffix");

  return (
    <Show when={tagSettings() !== undefined} fallback={<></>}>
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column">
          <Typography variant="h5">{headerTitle}</Typography>
          <Show
            when={tagSettings() !== null}
            fallback={
              <Typography variant="h6">
                リポジトリにイメージがありません
              </Typography>
            }
          >
            <TableContainer component={Paper}>
              <Table aria-label="simple table">
                <TableHead>
                  <TableRow>
                    <TableCell>設定</TableCell>
                    <TableCell>リリース対象イメージタグ</TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  <For each={tagSettings()} fallback={<></>}>
                    {(settingItem) => (
                      <TableRow>
                        <TableCell component="th" scope="row">
                          <Button
                            variant="contained"
                            size="small"
                            // color="primary"
                            onClick={() => {
                              openUriInNewTab(
                                `${uriPrefix}${settingItem.environment_name}${uriSuffix}`
                              );
                            }}
                            sx={{
                              textTransform: "none",
                              backgroundColor: `${buttonColor(
                                settingItem.environment_name
                              )}`,
                            }}
                            title={`${convertEnvironmentName(
                              settingItem.environment_name
                            )}環境のリリース設定画面を開く`}
                          >
                            {convertEnvironmentName(
                              settingItem.environment_name
                            )}
                            環境設定
                          </Button>
                        </TableCell>
                        <TableCell>{settingItem.tags.join(", ")}</TableCell>
                      </TableRow>
                    )}
                  </For>
                </TableBody>
              </Table>
            </TableContainer>
          </Show>
        </Stack>
      </Box>
    </Show>
  );
};
