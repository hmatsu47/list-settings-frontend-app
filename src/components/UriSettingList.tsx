import { For, Show } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Paper from "@suid/material/Paper";
import Table from "@suid/material/Table";
import TableBody from "@suid/material/TableBody";
import TableCell from "@suid/material/TableCell";
import TableContainer from "@suid/material/TableContainer";
import TableHead from "@suid/material/TableHead";
import TableRow from "@suid/material/TableRow";
import Typography from "@suid/material/Typography";
import { service, uriRemoteSettings, uriSettings } from "../signal";
import { formatDateTimeDisplay } from "../formatDateTime";
import { UriSetting } from "../type";
import { openUriInNewTab } from "../openUri";

const convertEnvironmentName = (environmentName: string) => {
  switch (environmentName) {
    case "stg":
      return "ステージング";
    case "prod":
      return "本番";
  }
  return environmentName;
};

export const UriSettingList = () => {
  const headerTitle = localStorage.getItem("uriSettingsHeaderTitle");

  return (
    <Show
      when={
        (service() && uriRemoteSettings() !== undefined) ||
        uriSettings() !== undefined
      }
      fallback={<></>}
    >
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Typography variant="h3">{headerTitle}</Typography>
        <Show
          when={uriRemoteSettings() !== null && uriSettings() !== null}
          fallback={
            <Typography variant="h6">
              リポジトリにイメージがありません
            </Typography>
          }
        >
          <TableContainer component={Paper}>
            <Table aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell>設定</TableCell>
                  <TableCell>イメージURI</TableCell>
                  <TableCell>リリース日時</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <SettingListParts settings={uriRemoteSettings()} />
                <SettingListParts settings={uriSettings()} />
              </TableBody>
            </Table>
          </TableContainer>
        </Show>
      </Box>
    </Show>
  );
};

type Props = {
  settings: UriSetting[] | null | undefined;
};

const SettingListParts = (props: Props) => {
  const uriPrefix = localStorage.getItem("uriSettingUriPrefix");
  const uriSuffix = localStorage.getItem("uriSettingUriSuffix");

  return (
    <Show when={props.settings} fallback={<></>}>
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <For each={props.settings} fallback={<></>}>
          {(settingItem) => (
            <Show
              when={settingItem.service_name === service()}
              fallback={<></>}
            >
              <TableRow>
                <TableCell component="th" scope="row">
                  <Button
                    variant="contained"
                    size="small"
                    color="primary"
                    onClick={() => {
                      openUriInNewTab(
                        `${uriPrefix}${settingItem.environment_name}${uriSuffix}`
                      );
                    }}
                    sx={{ textTransform: "none" }}
                    title={`${convertEnvironmentName(
                      settingItem.environment_name
                    )}環境のリリース設定画面を開く`}
                  >
                    {convertEnvironmentName(settingItem.environment_name)}
                    環境設定
                  </Button>
                </TableCell>
                <TableCell>
                  {settingItem.next_release
                    ? settingItem.next_release.image_uri
                    : settingItem.last_released.image_uri
                    ? settingItem.last_released.image_uri
                    : ""}
                </TableCell>
                <TableCell>
                  {settingItem.next_release
                    ? `次回：${formatDateTimeDisplay(
                        new Date(settingItem.next_release.release_at)
                      )}`
                    : settingItem.last_released
                    ? `前回：${formatDateTimeDisplay(
                        new Date(settingItem.last_released.released_at)
                      )}`
                    : ""}
                </TableCell>
              </TableRow>
            </Show>
          )}
        </For>
      </Box>
    </Show>
  );
};
