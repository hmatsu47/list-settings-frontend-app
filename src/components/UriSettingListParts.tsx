import { For, Show } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import TableCell from "@suid/material/TableCell";
import TableRow from "@suid/material/TableRow";
import { service } from "../signal";
import { formatDateTimeDisplay } from "../formatDateTime";
import { UriSetting } from "../type";
import { openUriInNewTab } from "../openUri";

const convertEnvironmentName = (environmentName: string) => {
  switch (environmentName) {
    case "stg":
      return "ステージング";
    case "prod":
      return "本番";
  }
  return environmentName;
};

type Props = {
  settings: UriSetting[] | null | undefined;
};

export const UriSettingListParts = (props: Props) => {
  const uriPrefix = localStorage.getItem("uriSettingUriPrefix");
  const uriSuffix = localStorage.getItem("uriSettingUriSuffix");

  return (
    <Show when={props.settings} fallback={<></>}>
      <For each={props.settings} fallback={<></>}>
        {(settingItem) => (
          <Show when={settingItem.service_name === service()} fallback={<></>}>
            <TableRow>
              <TableCell component="th" scope="row">
                <Button
                  variant="contained"
                  size="small"
                  color="primary"
                  onClick={() => {
                    openUriInNewTab(
                      `${uriPrefix}${settingItem.environment_name}${uriSuffix}`
                    );
                  }}
                  sx={{ textTransform: "none" }}
                  title={`${convertEnvironmentName(
                    settingItem.environment_name
                  )}環境のリリース設定画面を開く`}
                >
                  {convertEnvironmentName(settingItem.environment_name)}
                  環境設定
                </Button>
              </TableCell>
              <TableCell>
                {settingItem.next_release
                  ? `次回：${settingItem.next_release.image_uri}`
                  : settingItem.last_released
                  ? `前回：${settingItem.last_released.image_uri}`
                  : "（未設定）"}
              </TableCell>
              <TableCell>
                {settingItem.next_release
                  ? `次回：${formatDateTimeDisplay(
                      new Date(settingItem.next_release.release_at)
                    )}`
                  : settingItem.last_released
                  ? `前回：${formatDateTimeDisplay(
                      new Date(settingItem.last_released.released_at)
                    )}`
                  : "（未設定）"}
              </TableCell>
            </TableRow>
          </Show>
        )}
      </For>
    </Show>
  );
};
