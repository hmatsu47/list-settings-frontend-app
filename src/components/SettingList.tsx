import { onMount } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import {
  fetchLocal,
  fetchRemote,
  fetchUriSettings,
} from "../api/fetchUriSettings";
import { fetchTagSettings } from "../api/fetchTagSettings";
import { tagSettings, uriSettings, uriRemoteSettings } from "../signal";
import { TagSettingList } from "./TagSettingList";
import { UriSettingList } from "./UriSettingList";

export const SettingList = () => {
  onMount(() => {
    if (uriSettings() === undefined) {
      fetchUriSettings(fetchLocal);
    }
    if (uriRemoteSettings() === undefined) {
      fetchUriSettings(fetchRemote);
    }
    if (tagSettings() === undefined) {
      fetchTagSettings();
    }
  });

  return (
    <Stack spacing={2} direction="column">
      <UriSettingList />
      <Box
        sx={{
          padding: "10px 10px 0 10px",
          width: "100%",
          minWidth: "1024px",
          display: "flex",
          // justifyContent: "center",
        }}
        aria-live="polite"
      />
      <TagSettingList />
    </Stack>
  );
};
