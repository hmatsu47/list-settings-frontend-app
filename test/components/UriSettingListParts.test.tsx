import { describe, expect, test } from "vitest";
import { render } from "@solidjs/testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { UriSettingListParts } from "../../src/components/UriSettingListParts";
import { setService } from "../../src/signal";
import { LastReleased, NextRelease, UriSetting } from "../../src/type";

describe("<UriSettingListParts />", () => {
  const uriSettingList = [
    {
      title: "色設定なし・サービス1つ・環境1つ（本番）・リリースセットなし",
      colors: undefined,
      uriPrefix: "http://",
      uriSuffix: "-test.example.com",
      settings: [
        {
          environment_name: "prod",
          last_released: null,
          next_release: null,
          service_name: "test1",
        } as UriSetting,
      ],
      selectedService: "test1",
      expectedButtonLabel: ["本番環境設定"],
      expected: [{ label: "（未設定）", count: 2 }],
      expectedTitle: null,
    },
    {
      title:
        "サービス1つ・環境1つ（ステージング）・リポジトリにレコード1つ（過去リリースあり）",
      colors: '{"stg":"#00695f","prod":"#e91e63"}',
      uriPrefix: "http://",
      uriSuffix: "-stg.example.com",
      settings: [
        {
          environment_name: "stg",
          last_released: {
            image_uri: "hogefuga",
            released_at: new Date("2023-05-11T04:05:00+0900"),
          } as LastReleased,
          next_release: null,
          service_name: "test1",
        } as UriSetting,
      ],
      selectedService: "test1",
      expectedButtonLabel: ["ステージング環境設定"],
      expected: [
        { label: "前回：hogefuga", count: 1 },
        { label: "前回：2023-05-11 04:05", count: 1 },
      ],
      expectedTitle: null,
    },
    {
      title:
        "サービス1つ・環境1つ（本番）・リポジトリにレコード1つ（リリースセットあり）",
      colors: '{"stg":"#00695f","prod":"#e91e63"}',
      uriPrefix: "http://",
      uriSuffix: "-prod.example.com",
      settings: [
        {
          environment_name: "prod",
          last_released: null,
          next_release: {
            image_uri: "hogefuga",
            release_at: new Date("2023-05-15T04:05:00+0900"),
          } as NextRelease,
          service_name: "test1",
        } as UriSetting,
      ],
      selectedService: "test1",
      expectedButtonLabel: ["本番環境設定"],
      expected: [
        { label: "次回：hogefuga", count: 1 },
        { label: "次回：2023-05-15 04:05", count: 1 },
      ],
      expectedTitle: null,
    },
    {
      title:
        "サービス1つ・環境1つ（ステージング）・リポジトリにレコード1つ（過去リリースおよびリリースセットあり）",
      colors: '{"stg":"#00695f","prod":"#e91e63"}',
      uriPrefix: "http://",
      uriSuffix: "-stg.example.com",
      settings: [
        {
          environment_name: "stg",
          last_released: {
            image_uri: "foobar",
            released_at: new Date("2023-05-11T04:05:00+0900"),
          } as LastReleased,
          next_release: {
            image_uri: "hogefuga",
            release_at: new Date("2023-05-15T04:05:00+0900"),
          } as NextRelease,
          service_name: "test1",
        } as UriSetting,
      ],
      selectedService: "test1",
      expectedButtonLabel: ["ステージング環境設定"],
      expected: [
        { label: "次回：hogefuga", count: 1 },
        { label: "次回：2023-05-15 04:05", count: 1 },
      ],
      expectedTitle: ["前回：foobar | 2023-05-11 04:05"],
    },
    {
      title:
        "サービス2つ・環境1つ（本番）・リポジトリにレコード2つ（過去リリース：非表示およびリリースセット：表示あり）",
      colors: '{"stg":"#00695f","prod":"#e91e63"}',
      uriPrefix: "http://",
      uriSuffix: "-prod.example.com",
      settings: [
        {
          environment_name: "prod",
          last_released: {
            image_uri: "foobar",
            released_at: new Date("2023-05-11T04:05:00+0900"),
          } as LastReleased,
          next_release: null,
          service_name: "test1",
        } as UriSetting,
        {
          environment_name: "prod",
          last_released: null,
          next_release: {
            image_uri: "hogefuga",
            release_at: new Date("2023-05-15T04:05:00+0900"),
          } as NextRelease,
          service_name: "test2",
        } as UriSetting,
      ],
      selectedService: "test2",
      expectedButtonLabel: ["本番環境設定"],
      expected: [
        { label: "次回：hogefuga", count: 1 },
        { label: "次回：2023-05-15 04:05", count: 1 },
      ],
      expectedTitle: null,
    },
    {
      title:
        "サービス2つ・環境2つ（ステージング・本番）・リポジトリにレコード2つ（過去リリース：表示およびリリースセット：非表示あり）",
      colors: '{"stg":"#00695f","prod":"#e91e63"}',
      uriPrefix: "http://",
      uriSuffix: "-stg.example.com",
      settings: [
        {
          environment_name: "stg",
          last_released: {
            image_uri: "foobar",
            released_at: new Date("2023-05-11T04:05:00+0900"),
          } as LastReleased,
          next_release: null,
          service_name: "test1",
        } as UriSetting,
        {
          environment_name: "prod",
          last_released: null,
          next_release: null,
          service_name: "test1",
        } as UriSetting,
        {
          environment_name: "stg",
          last_released: null,
          next_release: null,
          service_name: "test2",
        } as UriSetting,
        {
          environment_name: "prod",
          last_released: null,
          next_release: {
            image_uri: "hogefuga",
            release_at: new Date("2023-05-15T04:05:00+0900"),
          } as NextRelease,
          service_name: "test2",
        } as UriSetting,
      ],
      selectedService: "test1",
      expectedButtonLabel: ["ステージング環境設定", "本番環境設定"],
      expected: [
        { label: "前回：foobar", count: 1 },
        { label: "前回：2023-05-11 04:05", count: 1 },
        { label: "（未設定）", count: 2 },
      ],
      expectedTitle: null,
    },
  ];
  beforeEach(() => {
    localStorage.removeItem("uriButtonColor");
    localStorage.removeItem("uriSettingUriPrefix");
    localStorage.removeItem("uriSettingUriSuffix");
  });
  afterEach(() => {
    localStorage.removeItem("selectedService");
  });
  uriSettingList.forEach((testCase) => {
    test(testCase.title, () => {
      if (testCase.colors) {
        localStorage.setItem("uriButtonColor", testCase.colors);
      }
      localStorage.setItem("uriSettingUriPrefix", testCase.uriPrefix);
      localStorage.setItem("uriSettingUriSuffix", testCase.uriSuffix);
      setService(testCase.selectedService);
      const { container, getByText, getAllByText, getByTitle, unmount } =
        render(() => <UriSettingListParts settings={testCase.settings} />);
      // css の名前が動的に変わるので固定値に置換
      const html = formatSnapshot(container.innerHTML);
      expect(html).toMatchSnapshot();
      // 各ボタン（クリックしない）
      testCase.expectedButtonLabel.forEach((buttonLabel) => {
        const text = getByText(buttonLabel) as HTMLElement;
        expect(text).toHaveTextContent(buttonLabel);
      });
      // 各 URI・日付
      testCase.expected.forEach((expected) => {
        if (expected.count === 1) {
          // URI と日付
          const text = getByText(expected.label) as HTMLElement;
          expect(text).toHaveTextContent(expected.label);
        } else {
          // （未設定）
          const text: string[] | any[] = getAllByText(
            expected.label
          ) as HTMLElement[];
          expect(text.length).toBe(expected.count);
        }
      });
      // title
      if (testCase.expectedTitle) {
        testCase.expectedTitle.forEach((title) => {
          const text = getByTitle(title) as HTMLElement;
          expect(text).toBeInTheDocument();
        });
      }
      unmount();
    });
  });
});
