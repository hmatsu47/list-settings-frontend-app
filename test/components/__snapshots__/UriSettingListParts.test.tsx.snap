// Vitest Snapshot v1

exports[`<UriSettingListParts /> > サービス1つ・環境1つ（ステージング）・リポジトリにレコード1つ（過去リリースあり） 1`] = `"<tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"ステージング環境のリリース設定画面を開く\\" type=\\"button\\">ステージング環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">前回：hogefuga</p></td><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">前回：2023-05-11 04:05</p></td></tr>"`;

exports[`<UriSettingListParts /> > サービス1つ・環境1つ（ステージング）・リポジトリにレコード1つ（過去リリースおよびリリースセットあり） 1`] = `"<tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"ステージング環境のリリース設定画面を開く\\" type=\\"button\\">ステージング環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：hogefuga<span class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall MuiIconButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" aria-label=\\"info\\" title=\\"前回：foobar | 2023-05-11 04:05\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSvgIcon-root css-xxxxxxxx\\" data-testid=\\"InfoIcon\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\"><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z\\"></path></svg><span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></span></p></td><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：2023-05-15 04:05</p></td></tr>"`;

exports[`<UriSettingListParts /> > サービス1つ・環境1つ（本番）・リポジトリにレコード1つ（リリースセットあり） 1`] = `"<tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"本番環境のリリース設定画面を開く\\" type=\\"button\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：hogefuga</p></td><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：2023-05-15 04:05</p></td></tr>"`;

exports[`<UriSettingListParts /> > サービス2つ・環境1つ（本番）・リポジトリにレコード2つ（過去リリース：非表示およびリリースセット：表示あり） 1`] = `"<tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"本番環境のリリース設定画面を開く\\" type=\\"button\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：hogefuga</p></td><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">次回：2023-05-15 04:05</p></td></tr>"`;

exports[`<UriSettingListParts /> > サービス2つ・環境2つ（ステージング・本番）・リポジトリにレコード2つ（過去リリース：表示およびリリースセット：非表示あり） 1`] = `"<tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"ステージング環境のリリース設定画面を開く\\" type=\\"button\\">ステージング環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">前回：foobar</p></td><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">前回：2023-05-11 04:05</p></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"本番環境のリリース設定画面を開く\\" type=\\"button\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">（未設定）</p></td><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">（未設定）</p></td></tr>"`;

exports[`<UriSettingListParts /> > 色設定なし・サービス1つ・環境1つ（本番）・リリースセットなし 1`] = `"<tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"本番環境のリリース設定画面を開く\\" type=\\"button\\">本番環境設定<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">（未設定）</p></td><td class=\\"MuiTableCell-root MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-root css-xxxxxxxx\\">（未設定）</p></td></tr>"`;
